src/app-components/Input/Input.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import type { InputHTMLAttributes, ReactNode } from 'react';
+
+import { Paragraph, Textfield } from '@digdir/designsystemet-react';
+import type { CharacterLimitProps } from '@digdir/designsystemet-react/dist/types/components/form/CharacterCounter';
+
+import classes from 'src/app-components/Input/Input.module.css';
+import type { InputType } from 'src/app-components/Input/constants';
+
+export type InputProps = {
+  size?: 'sm' | 'md' | 'lg';
+  prefix?: string;
+  suffix?: string;
+  characterLimit?: CharacterLimitProps;
+  error?: ReactNode;
+  disabled?: boolean;
+  id?: string;
+  readOnly?: boolean;
+  type?: InputType;
+  textOnly?: boolean;
+} & Pick<
+  InputHTMLAttributes<HTMLInputElement>,
+  | 'value'
+  | 'className'
+  | 'aria-label'
+  | 'aria-describedby'
+  | 'onChange'
+  | 'autoComplete'
+  | 'required'
+  | 'onBlur'
+  | 'placeholder'
+  | 'inputMode'
+  | 'style'
+>;
+
+export function Input(props: InputProps) {
+  const { size = 'sm', ...rest } = props;
+
+  if (props.textOnly) {
+    const { value, id, className } = props;
```

Comment:
It might be a bit misleading to just throw away all the other props here. I'm not sure where we should 'draw a line in the sand' when it comes to these pure components, but I would think this should be kept closer to the calling code rather than be a 'gotcha' inside here for future callers to stumble over.

Reply:
Not sure I agree with that. If we're following the same logic for everything, props like `hidden` or `disabled` would be exactly the same, yet it is a feature of that component to be able to be hidden or disabled based on other variables that might change. I don't see how `textOnly` is different 🤷‍♀️  
When it comes to the configuration of the layout component, it was already there and we have to ensure that that works as before.